File: advent-of-code/2015/02/01/solution.py
from pathlib import Path


def calculate_wrapping_paper(dimensions):
    """Calculate wrapping paper needed for a present."""
    l, w, h = map(int, dimensions.split('x'))
    
    # Calculate surface area
    surface_area = 2*l*w + 2*w*h + 2*h*l
    
    # Find smallest side
    sides = [l*w, w*h, h*l]
    slack = min(sides)
    
    return surface_area + slack

# Test cases with assertions
assert calculate_wrapping_paper("2x3x4") == 58
assert calculate_wrapping_paper("1x1x10") == 43

# Read input and calculate total
input_file = Path(__file__).parent.parent / "input.txt"
total = sum(calculate_wrapping_paper(line.strip()) for line in input_file.read_text().splitlines() if line.strip())
print(f"Python: {total}")

File: advent-of-code/2015/02/01/solution.cs
int CalculateWrappingPaper(string dimensions)
{
    var parts = dimensions.Split('x').Select(int.Parse).ToArray();
    var l = parts[0];
    var w = parts[1];
    var h = parts[2];
    
    // Calculate surface area
    var surfaceArea = 2*l*w + 2*w*h + 2*h*l;
    
    // Find smallest side
    var sides = new[] { l*w, w*h, h*l };
    var slack = sides.Min();
    
    return surfaceArea + slack;
}

// Test cases with assertions
System.Diagnostics.Debug.Assert(CalculateWrappingPaper("2x3x4") == 58);
System.Diagnostics.Debug.Assert(CalculateWrappingPaper("1x1x10") == 43);

// Read input and calculate total
var scriptDir = Directory.GetCurrentDirectory();
var inputPath = Path.Combine(scriptDir, "..", "input.txt");
var total = File.ReadAllLines(inputPath)
                .Where(line => !string.IsNullOrWhiteSpace(line))
                .Sum(CalculateWrappingPaper);

Console.WriteLine($"C#: {total}");
